### PR TITLE
pubsub: skip getProjectId for emulator usage

### LINKED ISSUE #2610: "Could not load the default credentials" when using PUBSUB_EMULATOR_HOST
#### Environment details

  - OS: Debian Jessie
  - Node.js version: 8.4.0
  - npm version: yarn 0.25.7
  - google-cloud-node/pubsub version: 0.14.1

#### Steps to reproduce

I am trying to use `gcloud beta emulators pubsub` locally for development. This used to work fine with `google-cloud/pubsub@0.13.2` and before, but since `0.14` I get authentication errors, see below.

Try any operation that sends requests to the api:

```js
const pubsub = require('@google-cloud/pubsub')();

console.log(`Emulator host: ${process.env.PUBSUB_EMULATOR_HOST}`);

pubsub.getTopics(function(err, topics) {
  console.log(err, topics);
});
```

Output:

```
$ node test.js

Emulator host: pubsub:8283
Error: Could not load the default credentials. Browse to https://developers.google.com/accounts/docs/application-default-credentials for more information.
    at /node_modules/google-auth-library/lib/auth/googleauth.js:316:21
    at /node_modules/google-auth-library/lib/auth/googleauth.js:346:7
    at Request._callback (/node_modules/google-auth-library/lib/transporters.js:70:30)
    at self.callback (/node_modules/request/request.js:188:22)
    at emitOne (events.js:115:13)
    at Request.emit (events.js:210:7)
    at Request.onRequestError (/node_modules/request/request.js:884:8)
    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at Socket.socketErrorListener (_http_client.js:401:9) undefined
```

With `@google-cloud/pubsub@0.13.2` it was easily possible to run against a local pubsub emulator without any credentials or whatsoever.

Is this intended behavior? What's the new recommended way of testing against local emulators? To me it feels unexpected to require credentials for running against the local emulator. Also what credentials would that be?!

I would rather expect that running locally against the emulator behaves the same as running on GCE, where all authentication happens automagically.

### LINKED ISSUE #2614: PubSub: Subscriptions fail to work with PubSub emulator
Using google-cloud/pubsub: 0.14.1

I'm trying to get PubSub working for offline development by using the emulator and creating a dummy keyfile. This works perfectly for creating topics, subscriptions and publishing messages... however it fails whenever the subscription activates its connection pool. Looks like the connection pool attempts to fetch metadata on the subscription from Google's servers and topics don't...

Here is a script you can run for yourselves below to reproduce the problem:

```javascript
'use strict';

const PubSubBuilder = require('@google-cloud/pubsub');
const UTIL = require('util');
const PATH = require('path');

/*
dummy-gcloud-keyfile.json:
{
  "type": "service_account",
  "project_id": "dummy-account",
  "private_key_id": "<Dummy private key here>",
  "private_key": "<Dummy key here>",
  "client_email": "dummy-key@foobar.com",
  "client_id": "<Dummy client id>",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://accounts.google.com/o/oauth2/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/dummy-key%40foobar.com"
}
*/

async function main() {
  // Listening to emulator on :8085
  const pubsub_client = PubSubBuilder({
    apiEndpoint: 'localhost:8085',
    projectId: 'foobar',
    keyFilename: PATH.resolve(__dirname, 'dummy-gcloud-keyfile.json'),
  });

  const topic = pubsub_client.topic('basic-test');
  const topic_exists = (await topic.exists())[0];
  if (!topic_exists) {
    console.log('Creating topic');
    await topic.create();
  }

  const subcription_name = 'basic-subscription';
  let subscription = topic.subscription(subcription_name);
  const subcription_exists = (await subscription.exists())[0];
  if (!subcription_exists) {
    console.log('Creating subscription');
    subscription =
        (await pubsub_client.createSubscription(topic, subcription_name))[0];
  }

  const publisher = topic.publisher();
  await publisher.publish(new Buffer('Test message'));
  console.log('Finished publishing messages');

  // NOTE: Everything works until here
  subscription.addListener('message', (message) => {
    console.log(message);
  });
}

main()
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```

### PR DESCRIPTION
**Pull request by callmehiphop**
Closes #2610 
Closes #2614

### CODE REVIEW

File: packages/pubsub/src/index.js [hunk 1]
```diff
@@ -293,6 +300,7 @@ PubSub.prototype.determineBaseUrl_ = function() {
   this.options.servicePath = baseUrlParts[0];
   this.options.port = baseUrlParts[1];
   this.options.sslCreds = grpc.credentials.createInsecure();
+  this.isEmulator = true;
```
Comment (stephenplusplus) [inline packages/pubsub/src/index.js]: I don't think there's a test for this.

File: packages/pubsub/src/index.js [hunk 2]
```diff
@@ -76,11 +82,12 @@ function PubSub(options) {
     libVersion: PKG.version
   }, options);
 
+  this.isEmulator = false;
```
Comment (stephenplusplus) [inline packages/pubsub/src/index.js]: Not tested.

Comment (callmehiphop) [reply packages/pubsub/src/index.js]: Whoops! Added

File: packages/pubsub/src/index.js (hunk 1 above)
Comment (callmehiphop) [reply packages/pubsub/src/index.js]: Done!

**Pull request merged**